starter-code/clue.js

```diff
@@ -1,92 +1,125 @@
-  image: https://metrouk2.files.wordpress.com/2016/07/miss-scarlett.jpg,
-  occupation: Actor
-
- Mrs Peacock
-  first name: Eleanor
-  last name: Peacock
-  color: blue
-  description: She is from a wealthy family and uses her status and money to earn popularity
-  age: 36
-  image: https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg
-  occupation: Socialité
-
- Mr Mustard
-  first name: Jack
-  last name: Mustard
-  color: yellow
-  description: He is a former football player who tries to get by on his former glory
-  age: 62
-  image: https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg
-  occupation: Retired Football player
-
----- Weapons ----
-
-Rope - Weight: 10
-Knife - Weight: 8
-Candlestick - Weight: 2
-Dumbbell - Weight: 30
-Poison - Weight: 2
-Axe - Weight: 15
-Bat - Weight: 13
-Trophy - Weight: 25
-Pistol - Weight: 20
-
-
----- Rooms ----
-Dinning Room
-Conservatory
-Kitchen
-Study
-Library
-Billiard Room
-Lounge
-Ballroom
-Hall
-Spa
-Living Room
-Observatory
-Theater
-Guest House
-Patio
-
-*/
+  image: "https://metrouk2.files.wordpress.com/2016/07/miss-scarlett.jpg",
+  occupation: "Actor"
+  }
+
+ var mrsPeacock = {
+  firstName: "Eleanor",
+  lastName: "Peacock",
+  color: "blue",
+  description: "She is from a wealthy family and uses her status and money to earn popularity",
+  age: 36,
+  image: "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+  occupation: "Socialité"
+ }
+
+ var mrMustard = {
+  firstName: "Jack",
+  lastName: "Mustard",
+  color: "yellow",
+  description: "He is a former football player who tries to get by on his former glory",
+  age: 62,
+  image: "https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg",
+  occupation: "Retired Football player"
+ }
+
+ var characters = [mrGreen, drOrchid, profPlum, missScarlet, mrsPeacock, mrMustard];
+
+
+
+
+var weapons = [
+
+{type: "Rope", weight: 10},
+{type: "Knife", weight: 8},
+{type: "Candlestick", weight:2},
+{type: "Dumbbell", weight: 30},
+{type: "Poison", weight: 2},
+{type: "Axe", weight: 15},
+{type: "Bat", weight: 13},
+{type: "Trophy", weight: 25},
+{type: "Pistol", weight: 20}
+];
+
+var rooms = [
+  "Dinning Room",
+  "Conservatory",
+  "Kitchen",
+  "Study",
+  "Library",
+  "Billiard Room",
+  "Lounge",
+  "Ballroom",
+  "Hall",
+  "Spa",
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio"
+]
+
+
+function random_selector(array) {
+  return Math.floor(Math.random()*array.length);
+}
+
+function pick_mistery(char, weap, rooms) {
+  return [
+    char[random_selector (char)],
+    weap[random_selector (weap)],
+    rooms[random_selector (rooms)]
+ ];
+ }
+
+function reveal_mistery(mistery) {
+  console.log("the character is: ", mistery[0], " with the weapon: ", mistery[1], "in the room: ", mistery[2]);
+}
+
+var mistery_envelope = pick_mistery(characters, weapons, rooms);
+reveal_mistery(mistery_envelope);
+
+
+// CODE GIVEN BELOW
+
+
 
 function random_selector(array) {
```

Comment:
Your original code would "work" if you get rid of these empty functions (lines 124, 128, 132). The problem is that you have the same functions twice and the second time they appea, they are empty. Anyways, there is a better approach to solve this problem, follow my comments.